lib/validates_overlap/overlap_validator.rb

```diff
@@ -98,11 +102,14 @@ def generate_overlap_sql_conditions(record)
     if record.new_record?
       self.sql_conditions = main_condition
     else
-      self.sql_conditions = "#{main_condition} AND #{record_table_name(record)}.#{primary_key_name} != #{record.send(primary_key_name)}"
+      if primary_key_value(primary_key_name, record).is_a?String
```

Comment:
Maybe something like:

```
        def primary_key_type
          column_type(model.primary_key).to_sym
        end
```

Idea stolen from https://github.com/globalize/globalize/pull/430/files since that gem needed to support UUID as well.